app/src/main/java/org/openobservatory/ooniprobe/activity/ResultDetailActivity.java

```diff
@@ -105,6 +110,54 @@ protected void onResume() {
         load();
     }
 
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.rerun, menu);
+        return super.onCreateOptionsMenu(menu);
+    }
+
+    @Override
+    public boolean onPrepareOptionsMenu(Menu menu) {
+        invalidateOptionsMenu();
+        if (!result.test_group_name.equals(WebsitesSuite.NAME))
+            menu.findItem(R.id.reRun).setVisible(false);
+        return super.onPrepareOptionsMenu(menu);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.reRun:
+                new ConfirmDialogFragment.Builder()
+                        .withExtra("re_run")
+                        //TODO sub these rows when the string is inserted
+                        //.withMessage(getString(R.string.Modal_ReRun_Websites_Title))
+                        //.withPositiveButton(getString(R.string.Modal_ReRun_Websites_Run))
+                        .withMessage(getString(R.string.Modal_DoYouWantToDeleteThisTest))
+                        .withPositiveButton(getString(R.string.Modal_Delete))
+                        .build().show(getSupportFragmentManager(), null);
+                return true;
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+    }
+
+    private void reRunTest() {
```

Comment:
Please choose a different name for this method that more clearly convey what it does. (Context: It is my understanding that this method is only used to rerun the websites nettest.)